src/crate/client/doctests/sqlalchemy.txt

```diff
@@ -0,0 +1,379 @@
+sub-queries yet.
+
+This means that queries with count have to be written in one of the following
+ways::
+
+    >>> session.query(sa.func.count(Character.id)).scalar()
+    2
+
+    >>> session.query(sa.func.count('*')).select_from(Character).scalar()
+    2
+
+Using the ``group_by`` clause is similar::
+
+    >>> session.query(sa.func.count(Character.id), Character.name) \
+    ...     .group_by(Character.name) \
+    ...     .order_by(sa.desc(sa.func.count(Character.id))) \
+    ...     .order_by(Character.name).all()
+    [(1, u'Arthur Dent'), (1, u'Tricia McMillan')]
+
+Fulltext Search with MATCH Predicate
+------------------------------------
+
+Fulltext Search in CrateDB is performed using the MATCH Predicate. The
+CrateDB SQLAlchemy driver comes with a ``match`` function in the
+``predicates`` namespace, which can be used to search on one or multiple
+fields.
+
+::
+
+    >>> from crate.client.sqlalchemy.predicates import match
+    >>> session.query(Character.name) \
+    ...     .filter(match(Character.name_ft, 'Arthur')) \
+    ...     .all()
+    [('Arthur Dent',)]
+
+
+To get the relevance of a matching row, an internal system column
+``_score`` can be selected. It's a numeric value which is relative to
+the other rows. The higher the score value, the more relevant the row.
+In most cases ``_score`` is not part of the SQLAlchemy Table definition,
+so it must be passed as a string::
+
+    >>> session.query(Character.name, '_score') \
+    ...     .filter(match(Character.quote_ft, 'space')) \
+    ...     .all()
+    [('Tricia McMillan', ...)]
+
+To search on multiple columns you have to pass a dictionary with columns
+and ``boost`` attached. ``boost`` is a factor that increases the
+relevance of a column in respect to the other columns::
+
+    >>> session.query(Character.name) \
+    ...           .filter(match({Character.name_ft: 1.5, Character.quote_ft: 0.1},
+    ...                         'Arthur')) \
+    ...     .order_by(sa.desc('_score')) \
+    ...     .all()
+    [('Arthur Dent',), ('Tricia McMillan',)]
+
+The match type determines how the query_term is applied and the ``_score`` is
+created, thus it influences which documents are considered more relevant.
+The default match_type is best_fields::
+
+    >>> session.query(Character.name) \
+    ...     .filter(
+    ...         match(Character.name_ft, 'Arth',
+    ...                 match_type='phrase',
+    ...                 options={'fuzziness': 3})
+    ...     ) \
+    ...     .all()
+    [(u'Arthur Dent',)]
+
+It's not possible to specify options without the match_type argument::
+
+    >>> session.query(Character.name) \
+    ...     .filter(
+    ...         match(Character.name_ft, 'Arth',
+    ...                 options={'fuzziness': 3})
+    ...     ) \
+    ...     .all()
+    Traceback (most recent call last):
+    ValueError: missing match_type. It's not allowed to specify options without match_type
+
+Create and Delete Tables
+------------------------
```

Comment:
could  be removed?

Reply:
I've kept this here because it seems like an important part of the tests, and a logical grouping of the two subsections. I have, however, changed the heading level